util/src/main/scala/Money.scala

```diff
@@ -343,33 +343,27 @@ object Money {
 }
 
 case class HighPrecisionMoney private (
-    amount: BigDecimal,
+    preciseAmount: Long,
     fractionDigits: Int,
     centAmount: Long,
     currency: Currency)
     extends BaseMoney
     with Ordered[Money] {
   import HighPrecisionMoney._
 
-  require(
-    amount.scale == fractionDigits,
-    "The scale of the given amount does not match the scale of the provided currency." +
-      " - " + amount.scale + " <-> " + fractionDigits
-  )
-
   require(
     fractionDigits >= currency.getDefaultFractionDigits,
     "`fractionDigits` should be  >= than the default fraction digits of the currency.")
 
   val `type`: String = TypeName
 
-  lazy val preciseAmountAsLong: Long =
-    (amount * Money.bdTen.pow(fractionDigits)).toLongExact // left side could be cached if necessary
+  lazy val amount: BigDecimal =
```

Comment:
Why `lazy` if we call for value few lines bellow?

Reply:
The idea is that many times HighPrecisionMoney is used without any calculations. Meaning for example we only get it from mongo and convert it to json or the other way around. 
In that case many of these methods won't be called and amount can stay uninitialized.